src/auth/users.controller.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Post,
+  ValidationPipe,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { CreateUserDto } from './input/create.user.dto';
+import { User } from './entities/user.entity';
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+
+@Controller('users')
+export class UsersController {
+  constructor(
+    private readonly authService: AuthService,
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+  ) {}
+
+  @Post()
+  async create(@Body(new ValidationPipe()) createUserDto: CreateUserDto) {
+    const user = new User();
+
+    if (createUserDto.password !== createUserDto.confirmPassword) {
+      throw new BadRequestException(['Passwords are not identical!']);
+    }
+
+    const existingUSer = await this.userRepository.findOne({
+      where: [
+        { username: createUserDto.userName },
+        { email: createUserDto.email },
+      ],
+    });
+
+    if (existingUSer) {
+      throw new BadRequestException(['username or email is already taken']);
+    }
+
+    user.username = createUserDto.userName;
+    user.password = await this.authService.hashPassword(createUserDto.password);
+    user.email = createUserDto.email;
+
+    return {
+      ...(await this.userRepository.save(user)),
```

Comment:
what do we get in the result of this?

Reply:
![image](https://user-images.githubusercontent.com/85223358/127631005-b2889245-70ff-4183-8111-4882af3e8a85.png)

Reply:
Done. The method doesn't return the user's password anymore. I've refactored auth controller.